include/engine/routing_algorithms/routing_base_ch.hpp

```diff
@@ -288,6 +288,106 @@ void unpackPath(const DataFacade<Algorithm> &facade,
     }
 }
 
+template <typename BidirectionalIterator>
+EdgeDistance calculateEBGNodeAnnotations(const DataFacade<Algorithm> &facade,
+                                         BidirectionalIterator packed_path_begin,
+                                         BidirectionalIterator packed_path_end)
+{
+    // Make sure we have at least something to unpack
+    if (packed_path_begin == packed_path_end ||
+        std::distance(packed_path_begin, packed_path_end) <= 1)
+        return 0;
+
+    std::stack<std::tuple<NodeID, NodeID, bool>> recursion_stack;
+    std::stack<EdgeDistance> distance_stack;
+    // We have to push the path in reverse order onto the stack because it's LIFO.
+    for (auto current = std::prev(packed_path_end); current != packed_path_begin;
+         current = std::prev(current))
+    {
+        recursion_stack.emplace(*std::prev(current), *current, false);
+    }
+
+    std::tuple<NodeID, NodeID, bool> edge;
```

Comment:
as we don't use the cache currently, we can remove `bool` (and then need to remove it from the following code as well)

Reply:
we made the decision to leave this in